src/config.rs

```diff
@@ -241,12 +241,12 @@ impl Config {
     fn sum_of_env_toml_values<'a>(
         &'a self,
         env_values: Option<Vec<String>>,
-        toml_getter: impl FnOnce(&'a CrossToml) -> &'a [String],
+        toml_getter: impl FnOnce(&'a CrossToml) -> Option<&'a [String]>,
     ) -> Result<Vec<String>> {
         let mut collect = vec![];
         if let Some(mut vars) = env_values {
             collect.append(&mut vars);
-        } else if let Some(toml_values) = self.toml.as_ref().map(toml_getter) {
+        } else if let Some(Some(toml_values)) = self.toml.as_ref().map(toml_getter) {
```

Comment:
very nice, buut

```suggestion
        } else if let Some(toml_values) = self.toml.as_ref().and_then(toml_getter) {
```

Reply:
Even better...

Reply:
This is a good oppurtunity for a clippy lint